api/client/commands.go

```diff
@@ -2183,7 +2183,7 @@ func (cli *DockerCli) createContainer(config *runconfig.Config, hostConfig *runc
 	//create the container
 	stream, statusCode, err := cli.call("POST", "/containers/create?"+containerValues.Encode(), mergedConfig, false)
 	//if image not found try to pull it
-	if statusCode == 404 {
+	if statusCode == 404 && strings.Contains(err.Error(), config.Image) {
```

Comment:
@estesp Yes, 404 is not specifically related to `image not found`, it's to simple to just check the `statusCode` , what do you think `if statusCode == 404 && strings.Contains(err.Error(), "No such image") && strings.Contains(err.Error(), config.Image)` here?